### PR TITLE
feat: newsletterbox block

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue

Closes # https://vsf.atlassian.net/jira/software/c/projects/SFUI2/boards/69?modal=detail&selectedIssue=SFUI2-664

# Scope of work

NewsletterBox block

In task there's a note that this should be in Banners but as this description was made before and there was already a separate place for NewsletterBox with thumbnail then it was put to this block

Two alerts are added separately (for demo purposes) as we can't do a component and there might be a case where we can see both of them (one after another)

# Screenshots of visual changes
Screenshot 2023-04-14 at 08 33 32
Screenshot 2023-04-14 at 08 33 24
Screenshot 2023-04-14 at 08 29 52
Screenshot 2023-04-14 at 08 29 01

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [x] cypress tests created

### CODE REVIEW

File: apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import { ShowcasePageLayout } from '../../showcases';
+// #region source
+import { useState } from 'react';
+import { SfButton, SfInput, SfLink, SfIconCheckCircle, SfIconClose } from '@storefront-ui/react';
+
+export default function NewsletterBox() {
+  const [inputValue, setInputValue] = useState('');
+  const [positiveAlert, setPositiveAlert] = useState(false);
+  const [errorAlert, setErrorAlert] = useState(false);
+  const [emailDataBase, setEmailDataBase] = useState<string[]>([]);
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx:10]: This is cool, however I think there should be info in docs description that if you pass twice same email you can see error alert, its not obvious without reading code

File: apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    if (!email) return;
+    if (checkEmailDataBase(email)) {
+      setErrorAlert(true);
+      setTimeout(() => setErrorAlert(false), 5000);
+    } else {
+      setPositiveAlert(true);
+      setEmailDataBase(() => [...emailDataBase, email]);
+      setTimeout(() => setPositiveAlert(false), 5000);
+    }
+    console.log(email);
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx:25]: leftover

File: apps/preview/nuxt/pages/showcases/NewsletterBox/NewsletterBox.vue [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  if (!email) return;
+  if (checkEmailDataBase(email)) {
+    showErrorAlert.value = true;
+    setTimeout(() => (showErrorAlert.value = false), 5000);
+  } else {
+    showPositiveAlert.value = true;
+    emailDataBase.value.push(email);
+    setTimeout(() => (showPositiveAlert.value = false), 5000);
+  }
+  console.log(email);
```
Comment (Szymon-dziewonski) [inline apps/preview/nuxt/pages/showcases/NewsletterBox/NewsletterBox.vue:81]: leftover

File: apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx (hunk 1 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx:10]: sure, I'll add a description for this

File: apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx (hunk 2 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx:25]: there is a point in a task that email should appear in console but I can remove it form the code :)

Comment (Szymon-dziewonski) [reply apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx:25]: then we can keep it :)

File: apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx (hunk 1 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/NewsletterBox/NewsletterBox.tsx:10]: https://github.com/vuestorefront/storefront-ui/pull/2595/commits/a723fb52c68eaf4658fe01dbada8e0e305892a7d

**Review by Szymon-dziewonski — APPROVED** [review]
perfect ! :)

**Pull request merged**